### DISCUSSION TITLE
What are the top 10 highest priority vLEI related KERI issues?

### DISCUSSION THREAD

**Discussion by 2byrds** [WebOfTrust/keri#77] [Ideas]
Please respond here with any issue (no matter how old) from a KERI-related repo that you would like highlighted at the KERI dev meeting on Thursdays. To gamify it we will highlight the issue at the meeting and then discuss if it belongs in the top 10 vLEI related issues.

**Comment by 2byrds** [comment]
Here is a recent list from @daviddm and @lenkan that we can consider:

https://github.com/WebOfTrust/signify-ts/issues/227
https://github.com/WebOfTrust/keria/issues/175
https://github.com/WebOfTrust/keria/issues/201
https://github.com/WebOfTrust/keria/issues/184
https://github.com/WebOfTrust/keria/issues/37
https://github.com/WebOfTrust/keria/issues/222
https://github.com/WebOfTrust/keria/issues/171
https://github.com/WebOfTrust/keria/issues/200
https://github.com/WebOfTrust/keria/issues/217

**Comment by kentbull** [comment]
Backup and Restore (https://github.com/WebOfTrust/keria/issues/37) is the highest priority issue to me. Everything else is secondary. We must have a good disaster recovery story.

A good second, of the issues listed above, is Graceful Shutdown (https://github.com/WebOfTrust/keria/issues/175).

And a third would be Passcode Rotation (https://github.com/WebOfTrust/keria/issues/222).

As I have time I will volunteer to help. Graceful shutdown is the one I most likely could tackle in the near future. We just need to signal to the Doist instance in `directing.runController` to drain existing connections and shut the event loop down.

**Comment by lenkan** [comment]
I have created two new issues that is blocking us:

- https://github.com/WebOfTrust/keripy/issues/762
- https://github.com/WebOfTrust/keria/issues/231

We have not been able to create proper reproductions yet.

**Comment by 2byrds** [comment]
Tired of searching for lost events, this will help immensely! https://github.com/WebOfTrust/keripy/issues/763

**Comment by 2byrds** [comment]
Proposing we discuss KERIA performance as there are at least two issues now and @nkongsuwan and I discussed the importance of KERIA being stable/predictible at IIW:
https://github.com/WebOfTrust/keria/issues/232
KERIA high CPU usage makes longer term testing difficult and is very costly in production. One medium term goal is to make KERIA instances efficient. It would also good to discuss what we 'guess' a reasonable number of simultaneous agents should be while idle, with normal usage, and beyond.

**Comment by 2byrds** [comment]
Consolidate/Updated discussion here https://github.com/WebOfTrust/keri/discussions/78
